src/octoprint/util/__init__.py

```diff
@@ -1090,6 +1090,70 @@ def _finish(self, *callbacks):
 		if callable(self.on_finish):
 			self.on_finish()
 
+class ResettableTimer(threading.Thread):
+	"""
+	This class represents an action that should be run after a specified amount of time. It is similar to python's
+	own :class:`threading.Timer` class, with the addition of being able to reset the counter to zero.
+
+	ResettableTimers are started, as with threads, by calling their ``start()`` method. The timer can be stopped (in
+	between runs) by calling the :func:`cancel` method. Resetting the counter can be done with the :func:`reset` method.
+
+	For example:
+
+	.. code-block:: python
+
+	   def hello():
+	       print("Ran hello() at {}").format(time.time())
+
+	   t = ResettableTimers(60.0, hello)
+	   t.start()
+	   print("Started at {}").format(time.time())
+	   time.sleep(30)
+	   t.reset()
+	   print("Reset at {}").format(time.time())
+
+	Arguments:
+	    interval (float or callable): The interval before calling ``function``, in seconds. Can also be a callable
+	        returning the interval to use, in case the interval is not static.
+	    function (callable): The function to call.
+	    args (list or tuple): The arguments for the ``function`` call. Defaults to an empty list.
+	    kwargs (dict): The keyword arguments for the ``function`` call. Defaults to an empty dict.
+	"""
+	
+	def __init__(self, interval, function, args=[], kwargs={}):
```

Comment:
The default arguments `args` and `kwargs` should not be mutable. Rather declare them with default value `None`, test for that and set them to empty list/dict if required:

``` python
def __init__(self, interval, function, args=None, kwargs=None):
    if args is None:
        args = []
    if kwargs is None:
        kwargs = dict()
```

---

Now that I look at this, these days I'd actually do something like 

``` python
def __init__(self, interval, function, *args, **kwargs):
```

here - that way you'd circumvent the mutable default values and make the initialization of a timer nicer to read:

``` python
def hello(who, suffix="!"):
    print("Hello {}{}".format(who, suffix)

timer = ResettableTimer(60.0, hello, "World", suffix="?")
```

But that of course wouldn't match `RepeatedTimer`. Meh.